tests/test_dataset.py

```diff
@@ -218,3 +222,37 @@ def _check_row_equality(d1, d2):
 
     _check_row_equality(dataset[0], {"A": root["A"][0, :], "B": root["B"][0, :]})
     _check_row_equality(dataset[10], {"A": root["A"][10, :], "B": root["B"][10, :]})
+
+
+def test_missing_codec_error(tmpdir):
```

Comment:
**Minor:** I noticed that we’re currently mixing the use of `tmp_path` and `tmpdir`. While it doesn’t need to be addressed in this PR, it seems like `tmp_path` is nowadays [the preferred way](https://docs.pytest.org/en/stable/how-to/tmp_path.html#the-tmpdir-and-tmpdir-factory-fixtures) over `tmpdir`.

Reply:
Good catch!